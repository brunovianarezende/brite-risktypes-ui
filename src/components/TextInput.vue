<template>
  <el-form-item :label="label">
    <el-input
      :placeholder="placeholder"
      :value="value"
      @input="value => { $emit('input', value) }"
    ></el-input>
  </el-form-item>
</template>

<script>
export default {
  name: 'TextInput',
  props: ['value', 'label', 'placeholder']
}
</script>
