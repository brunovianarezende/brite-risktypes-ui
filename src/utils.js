import Vue from 'vue'
import ElementUI from 'element-ui'
import locale from 'element-ui/lib/locale/lang/en'
import VModal from 'vue-js-modal'

export const registerGlobalComponents = () => {
  Vue.use(ElementUI, { locale })
  Vue.use(VModal)
}
