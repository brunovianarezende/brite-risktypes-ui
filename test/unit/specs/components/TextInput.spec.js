import Vue from 'vue'
import TextInput from '@/components/TextInput'
import {registerGlobalComponents} from '@/utils'

registerGlobalComponents()

describe('TextInput.vue', () => {
  describe('interface', () => {
    it('should render correctly', () => {
      const Comp = {
        render (createElement) {
          return createElement(
            'el-form', {}, [
              createElement(TextInput,
                {props: {label: this.label, placeholder: this.placeholder}})
            ]
          )
        },
        props: ['label', 'placeholder'],
        components: {TextInput},
        propsData: {
          label: 'The label',
          placeholder: 'The placeholder'
        }
      }

      const vm = new Vue(Comp).$mount()
      return Vue.nextTick()
        .then(() => {
          expect(vm.$el).toMatchSnapshot()
        })
    })

    it('should support v-model', () => {
      const Comp = {
        render (createElement) {
          const div = createElement('div', this.value)
          const textInput = createElement(
            TextInput,
            {
              props: {
                label: this.label,
                placeholder: this.placeholder,
                value: this.value
              },
              on: {
                input: (v) => { this.value = v }
              }
            })

          return createElement(
            'el-form', {}, [div, textInput]
          )
        },
        props: ['label', 'placeholder'],
        data: {
          value: 'Hi'
        },
        components: {TextInput},
        propsData: {
          label: 'The label',
          placeholder: 'The placeholder'
        }
      }

      const vm = new Vue(Comp).$mount()
      return Vue.nextTick()
        .then(() => {
          const input = vm.$el.querySelector('input')
          expect(input.value).toEqual('Hi')
          input.value = 'hello'
          input.dispatchEvent(new Event('input'))
          return Vue.nextTick()
        })
        .then(() => {
          expect(vm.$el).toMatchSnapshot()
        })
    })
  })
})
