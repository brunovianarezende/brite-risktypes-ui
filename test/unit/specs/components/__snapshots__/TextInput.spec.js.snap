// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput.vue interface should render correctly 1`] = `
<form
  class="el-form"
>
  <div
    class="el-form-item"
  >
    <label
      class="el-form-item__label"
    >
      The label
    </label>
    <div
      class="el-form-item__content"
    >
      <div
        class="el-input"
      >
        <!---->
        <input
          autocomplete="off"
          class="el-input__inner"
          placeholder="The placeholder"
          rows="2"
          type="text"
          validateevent="true"
        />
        <!---->
        <!---->
        <!---->
      </div>
      <!---->
    </div>
  </div>
</form>
`;

exports[`TextInput.vue interface should support v-model 1`] = `
<form
  class="el-form"
>
  <div>
    hello
  </div>
  <div
    class="el-form-item"
  >
    <label
      class="el-form-item__label"
    >
      The label
    </label>
    <div
      class="el-form-item__content"
    >
      <div
        class="el-input"
      >
        <!---->
        <input
          autocomplete="off"
          class="el-input__inner"
          placeholder="The placeholder"
          rows="2"
          type="text"
          validateevent="true"
        />
        <!---->
        <!---->
        <!---->
      </div>
      <!---->
    </div>
  </div>
</form>
`;
